pandas/_libs/lib.pyx

```diff
@@ -2247,6 +2258,28 @@ def maybe_convert_numeric(
     cdef:
         object val = values[0]
 
+    # Convert python strings into ones readable by C
+
+    cdef char* tsep
+    cdef char* dsep
+    # Use null char to represent lack of separator
+    if thousands is None:
+        tsep = "\0"
+    else:
+        bytes_tsep = thousands.encode("UTF-8")
+        tsep = bytes_tsep
+
+    bytes_dsep = decimal.encode("UTF-8")
+    dsep = bytes_dsep
+
+    # Validate separators
+    if len(tsep) > 1:
```

Comment:
Not sure why this would be useful but raising like this prevents a multi-byte character from being used as a separator. That should be tested either way

Reply:
Using wide characters for these separators does indeed seem pretty odd and an unlikely use case, if we do need to support this then it would require a change of approach as the C parser uses a `char` to represent the thousasnds separator.